File: SubtreeSum/solve/python/SubtreeSumSolver.py
#! /usr/bin/env python
import os, sys
import SubtreeSum

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)

        p = reader.next("int[]")
        reader.next()
        x = reader.next("int[]")

    result = SubtreeSum.getSum(p, x)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "int"))

if __name__ == "__main__":
    init()
    main()

File: SubtreeSum/solve/ruby/SubtreeSumSolver.rb
#! /usr/bin/env ruby

require_relative "SubtreeSum"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  p = reader.next(TArray.new(TInt))
  reader.next()
  x = reader.next(TArray.new(TInt))

  result = SubtreeSum.new().getSum(p, x)
  IO.write(ARGV[1], Writer.new.next(result, TInt).to_s)
end

init
main
